### DISCUSSION TITLE
UI Improvement: New Color Picker to allow textured/silk/metallic/transparent filaments and filament type overlay text in UI.

### DISCUSSION THREAD

**Discussion by michaeljsmalley** [OrcaSlicer/OrcaSlicer#3884] [Ideas]
My suggestion here is all about improving the Orca Slicer user experience. Most filaments come in pretty basic colors that are easy to represent in Orca Slicer's default color picker (I believe it just uses the OS-native selector). However, there exist hundreds of different varieties of very popular metallic, silk/iridescent, transparent, or textured filaments such as the following:

Printed Solid Vanilla Bean PLA
![45d7k4kd27y91](https://github.com/SoftFever/OrcaSlicer/assets/293608/edcb6cbd-65d5-41bb-8d64-a0ed3dd9496b)
(Source: https://www.reddit.com/media?url=https%3A%2F%2Fi.redd.it%2F45d7k4kd27y91.jpg)

![24b5f953-0db5-4cc1-ba9b-7fedf0eb311f __CR0,210,1000,619_PT0_SX970_V1___](https://github.com/SoftFever/OrcaSlicer/assets/293608/d4b1f0b3-8adc-42e5-9b8e-bc914d14553a)
(Source: https://www.amazon.com/Inland-Dual-Color-Filament-Dimensional/dp/B0BFFYMRL3)

These filaments are impossible to accurately represent using a solid color in Orca Slicer. Silk blue/green becomes a "turquoise" approximation. Metallic gold filaments become a flat gold or and silver a flat gray color. Textured filaments too are approximated to "gray". Effectively, the user is forced to choose the best solid color to represent their real-world filaments. It would be a huge leap in accuracy and the overall user experience if Orca Slicer's color picker were upgraded to support:

a) **Textures**: Imagine being able to select the actual texture of your filament, and then have it represented in the UI. Thinking even bigger, these may be usable in the renderer while in the Preparation pane (or as a special option under the Color scheme dropdown in the Preview pane.

b) **Texture builder**: If we don't want to deal with uploading images to the slicer, we could make a renderer that approximates the texture of the filament we are using. For example, if I were using the above Vanilla Bean PLA, I could choose a base color of gray, then say it's speckled and choose black at a specific frequency, and let the slicer generate a texture dynamically. I imagine a noise generator would do this job nicely. For iridescent textures like the silk PLA above, we may be able to select a few colors, render a gradient, then have a little shimmer effect over them (animated or still) to indicate it's silk/iridescent. For transparent filaments, maybe a similar effect would be used, where it looks frosted or like a light is shining through it. This could all be done via generators.

These features would make the representation of filaments in Orca Slicer more closely aligned with what is in our printers.

One last nice thing that could be implemented if this change were developed would be to have a label on the color itself indicating the plastic type (e.g. PLA, PETG, ABS, TPU, etc). This would make for a better user experience as well, allowing for much quicker identification of what is in a printer.

**Comment by einherrmerlin** [comment]
this needs to be a thing!

**Comment by inertiacreeping** [comment]
Agreed - would love this!

When printing designs in twin-colour filament which need to mate together (for instance, a cap screwing down on a bottle) it would be a HUGE help to visualise how the twin colours line up in the slicer
